src/__internal__/label/label.test.tsx

```diff
@@ -0,0 +1,166 @@
+
+  expect(screen.getByRole("tooltip", { name: "bar" })).toBeVisible();
+});
+
+test("Help icon is focusable", async () => {
+  const user = userEvent.setup();
+  render(<Label help="bar">foo</Label>);
+
+  const help = screen.getByRole("button", { name: "help" });
+  await user.click(help);
+
+  expect(help).toHaveFocus();
+
+  await user.tab();
+
+  expect(help).not.toHaveFocus();
+});
+
+test("renders Help tooltip with provided `tooltipId`", async () => {
+  const user = userEvent.setup();
+  render(
+    <Label help="bar" tooltipId="baz">
+      foo
+    </Label>
+  );
+
+  await user.hover(screen.getByRole("button", { name: "help" }));
+  const tooltip = screen.getByRole("tooltip", { name: "bar" });
+
+  expect(tooltip).toBeVisible();
+  expect(tooltip).toHaveAttribute("id", "baz");
+});
+
+test("renders label with provided `labelId`", () => {
+  render(<Label labelId="baz">foo</Label>);
+
+  expect(screen.getByText("foo")).toHaveAttribute("id", "baz");
+});
+
+test("renders with provided `aria-label`", () => {
+  render(<Label aria-label="baz">foo</Label>);
+
+  expect(screen.getByText("foo")).toHaveAccessibleName("baz");
+});
+
+test("does not render validation icon when `disabled` prop is true", () => {
+  render(
+    <Label error="error" disabled>
+      foo
+    </Label>
+  );
+
+  expect(screen.queryByTestId("icon-error")).not.toBeInTheDocument();
+});
+
+test("renders with provided `htmlFor`", () => {
+  render(<Label htmlFor="baz">foo</Label>);
+
+  expect(screen.getByText("foo")).toHaveAttribute("for", "baz");
+});
+
+test("does not render with `htmlFor` when not rendered as a `<label>` element", () => {
+  render(
+    <Label as="span" htmlFor="baz">
+      foo
+    </Label>
+  );
+
+  expect(screen.getByText("foo")).not.toHaveAttribute("for");
+});
+
+test.each(["error", "warning", "info"])(
+  "renders %s tooltip with position set to 'top' when label is inline",
+  async (validationProp) => {
+    const user = userEvent.setup();
+    const useFloatingSpy = jest.spyOn(floatingUi, "useFloating");
+    render(
+      <Label inline {...{ [validationProp]: "Message" }}>
+        foo
+      </Label>
+    );
+
+    await user.hover(screen.getByTestId(`icon-${validationProp}`));
+
+    expect(useFloatingSpy).toHaveBeenCalledWith(
+      expect.objectContaining({ placement: "top" })
+    );
+
+    useFloatingSpy.mockRestore();
+  }
+);
+
+test.each(["error", "warning", "info"])(
+  "renders %s tooltip with position set to 'right' when label is not inline",
+  async (validationProp) => {
+    const user = userEvent.setup();
+    const useFloatingSpy = jest.spyOn(floatingUi, "useFloating");
+    render(<Label {...{ [validationProp]: "Message" }}>foo</Label>);
+
+    await user.hover(screen.getByTestId(`icon-${validationProp}`));
+
+    expect(useFloatingSpy).toHaveBeenCalledWith(
+      expect.objectContaining({ placement: "right" })
+    );
+
+    useFloatingSpy.mockRestore();
```

Comment:
**suggestion**: Similar to the previous test, what do you think about this change? It would avoid the need to mock @floating-ui/react-dom:

```suggestion
    render(<Label {...{ [validationProp]: "Message" }}>foo</Label>);

    await user.hover(screen.getByTestId(`icon-${validationProp}`));

    expect(await screen.findByRole("tooltip")).toHaveAttribute("data-placement", "right");
```